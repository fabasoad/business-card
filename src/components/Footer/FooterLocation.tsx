import * as React from 'react'
import { useTranslation } from 'react-i18next'

export default function FooterLocation() {
  const { t } = useTranslation()
  return (
    <div className="col">
      <i className="fa fa-map-marker-alt fa-2x"></i>
      <p>{t('business-card-contact-city')}</p>
    </div>
  )
}
