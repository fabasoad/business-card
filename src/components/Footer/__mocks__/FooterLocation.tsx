import * as React from 'react'

export default function FooterLocation() {
  return (
    <div data-testid="FooterLocation"></div>
  )
}
