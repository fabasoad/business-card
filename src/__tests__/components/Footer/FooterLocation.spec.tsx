import '@testing-library/jest-dom'
import * as React from 'react'
import { render } from '@testing-library/react'

import FooterLocation from '../../../components/Footer/FooterLocation'

export function testFooterLocation(div: HTMLDivElement) {
  expect(div).toHaveAttribute('class', 'col')
  const i = div.querySelector('i');
  ['fa', 'fa-map-marker-alt', 'fa-2x'].forEach((c: string) => expect(i).toHaveClass(c))
  expect(div.querySelector('p')).toHaveTextContent('business-card-contact-city')
}

test('should render FooterLocation correctly', () => {
  const { container } = render(<FooterLocation />)
  testFooterLocation(container.querySelector('div'))
})
