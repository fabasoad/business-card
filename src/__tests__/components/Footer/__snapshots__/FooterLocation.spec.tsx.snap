// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FooterLocation correctly 1`] = `
<div
  className="col"
>
  <i
    className="fa fa-map-marker-alt fa-2x"
  />
  <p>
    business-card-contact-city
  </p>
</div>
`;
